include/zephyr/drivers/firmware/scmi/clk.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief SCMI clock protocol helpers
+ */
+
+#ifndef _INCLUDE_ZEPHYR_DRIVERS_FIRMWARE_SCMI_CLK_H_
+#define _INCLUDE_ZEPHYR_DRIVERS_FIRMWARE_SCMI_CLK_H_
+
+#include <zephyr/drivers/firmware/scmi/protocol.h>
+
+#define SCMI_CLK_CONFIG_DISABLE_ENABLE_MASK GENMASK(1, 0)
+#define SCMI_CLK_CONFIG_ENABLE_DISABLE(x)\
+	((uint32_t)(x) & SCMI_CLK_CONFIG_DISABLE_ENABLE_MASK)
+
+#define SCMI_CLK_ATTRIBUTES_CLK_NUM(x) ((x) & GENMASK(15, 0))
+
+/**
+ * @struct scmi_clock_config
+ *
+ * @brief Describes the parameters for the CLOCK_CONFIG_SET
+ * command
+ */
+struct scmi_clock_config {
+	uint32_t clk_id;
+	uint32_t attributes;
+	uint32_t extended_cfg_val;
+};
+
+/**
+ * @brief Clock protocol command message IDs
+ */
+enum scmi_clock_message {
+	SCMI_CLK_MSG_PROTOCOL_VERSION = 0x0,
+	SCMI_CLK_MSG_NEGOTIATE_PROTOCOL_VERSION = 0x10,
```

Comment:
Put protocol 0x10 to be after 0xf?

Reply:
thanks, will have these ordered by ID value